naps/cores/peripherals/bitbang_spi.py

```diff
@@ -0,0 +1,55 @@
+from nmigen import *
+
+from naps.cores.peripherals.mmio_gpio import MmioGpio
+from naps.soc.devicetree_overlay import devicetree_overlay
+from naps.soc.soc_platform import SocPlatform
+
+__all__ = ["BitbangSPI"]
+
+
+class BitbangSPI(Elaboratable):
+    def __init__(self, pins, name_suffix=""):
+        self.devicetree_name = "bitbang_spi" + name_suffix
+        self.pins = pins
+
+    def elaborate(self, platform: SocPlatform):
+        m = Module()
+
+        # for some reason, despite CS being specified as active high in the
+        # device tree, linux and/or the python spidev module treat it as active
+        # low. thus we invert CS here to make the pin active high as defined
+        pins = Record.like(self.pins)
```

Comment:
why add that level of indirection to all pins?

Reply:
good point, fixed